src/query_plan/operation.rs

```diff
@@ -491,32 +537,63 @@ pub(crate) enum NormalizedSelectionKey {
         /// directives applied on the field
         directives: Arc<DirectiveList>,
     },
-    FragmentSpread {
-        /// The fragment name referenced in the spread.
-        name: Name,
+    Fragment {
+        /// The optional type condition of the fragment.
+        type_condition: Option<Name>,
```

Comment:
Right, the key intentionally does not differentiate between fragment spreads and inline fragments. That's how it worked in JS and contains() relies on it. To be honest, I'm not entirely sure what the implications are of this in the Rust codebase, I should triple-check that in case it's introducing a bug elsewhere. Good question!

Reply:
Ahh there *is* a difference in the JS code between named and inline fragments!